NRC/yangswei_85_nrc_intensity.ipynb

```diff
@@ -0,0 +1,355 @@
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# predict label by NRC\n",
+    "yangswei_85_test['label_nrc'] = yangswei_85_test['text'].apply(lambda x: label_text(x, nrc_lexicon))\n",
+    "\n",
+    "yangswei_85_test.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "a3f458c0-ef02-4e30-a2d0-e7562b4fc5a3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "yangswei_85_test.to_csv('yangswei_85_nrc_inten_test.csv')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "857c07f9-5cd7-4ba7-804e-94cd392b7511",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "label_nrc\n",
+       "trust           7154\n",
+       "joy             5202\n",
+       "anticipation    3451\n",
+       "surprise        1928\n",
+       "sadness         1730\n",
+       "fear            1534\n",
+       "anger            858\n",
+       "disgust          432\n",
+       "Name: count, dtype: int64"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "label_counts = yangswei_85_test['label_nrc'].value_counts()\n",
+    "label_counts"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "a5505fc4-983f-48dd-9a01-6544c68eefe6",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Metrics for NRC:\n",
+      "Accuracy: 0.2145\n",
+      "Precision: 0.5461\n",
+      "Recall: 0.2145\n",
+      "F1-Score: 0.3038\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "C:\\Users\\minhd\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\site-packages\\sklearn\\metrics\\_classification.py:1531: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
+      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n",
+      "C:\\Users\\minhd\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\site-packages\\sklearn\\metrics\\_classification.py:1531: UndefinedMetricWarning: Recall is ill-defined and being set to 0.0 in labels with no true samples. Use `zero_division` parameter to control this behavior.\n",
+      "  _warn_prf(average, modifier, f\"{metric.capitalize()} is\", len(result))\n"
+     ]
+    }
+   ],
+   "source": [
+    "from sklearn.metrics import accuracy_score, precision_recall_fscore_support\n",
+    "y_true = yangswei_85_test['label']\n",
+    "y_pred = yangswei_85_test['label_nrc']\n",
+    "\n",
+    "accuracy = accuracy_score(y_true, y_pred)\n",
+    "precision, recall, f1, _ = precision_recall_fscore_support(y_true, y_pred, average='weighted')\n",
+    "\n",
+    "# show the results\n",
```

Comment:
add code that saves the results to a file, from the notebook there is no automation regarding the saving of files.